dkulib/parallelizer/parallelizer.py

```diff
@@ -126,62 +149,46 @@ def _get_unique_output_column_names(self, existing_names: List[AnyStr]) -> Named
         )
 
     def _apply_function_and_parse_response(
-        self, row: Dict = None, batch: List[Dict] = None, **function_kwargs,
+        self, batch: List[Dict] = None, **function_kwargs,
     ) -> Union[Dict, List[Dict]]:  # sourcery skip: or-if-exp-identity
         """Wrap a row-by-row or batch function with error logging and response parsing
-
         It applies `self.function` and:
         - If batch, parse the function response to extract results and errors using `self.batch_response_parser`
         - handles errors from the function with two methods:
             * (default) log the error message as a warning and return the row with error keys
             * fail if there is an error (if `self.error_handling == ErrorHandling.FAIL`)
-
         """
-        if row and batch:
-            raise (ValueError("Please use either row or batch as arguments, but not both"))
-        output = deepcopy(row) if row else deepcopy(batch)
+        output = deepcopy(batch)
         for output_column in self._output_column_names:
-            if row:
-                output[output_column] = ""
-            else:
-                for output_row in output:
-                    output_row[output_column] = ""
+            for output_row in output:
+                output_row[output_column] = ""
         try:
-            response = (
-                self.function(row=row, **function_kwargs) if row else self.function(batch=batch, **function_kwargs)
-            )
-            if row:
-                output[self._output_column_names.response] = response
+            if self.batch_support is False:
+                # In the row-by-row case, there is only one element in the list as batch_size=1
+                response = (
+                    self.function(row=batch[0], **function_kwargs)
+                )
             else:
-                output = self.batch_response_parser(
-                    batch=batch, response=response, output_column_names=self._output_column_names
+                response = (
+                    self.function(batch=batch, **function_kwargs)
                 )
-                errors = [
-                    row[self._output_column_names.error_message]
-                    for row in output
-                    if row[self._output_column_names.error_message]
-                ]
-                if errors:
-                    raise BatchError(str(errors))
```

Comment:
These lines should be kept, otherwise, error logging will not work in the batch case

Reply:
Sorry added them back in!